### ISSUE TITLE
Aligment for i32, i64, f32, f64 is more than EABI states

### ISSUE THREAD

**Issue by chbessonova** [access-softek/msp430-clang#24]
EABI p.2.1 states that char type has 8 bit alignment, and all other - 16 bit alignment.

Here is a clang generated IR for all basic types:
```
target datalayout = "e-m:e-p:16:16-i32:16-i64:16-f32:16-f64:16-a:8-n8:16-S16"
target triple = "msp430"
@c = global i8 1, align 1
@s = global i16 266, align 2
@t = global i16 266, align 2
@l = global i32 163990, align 2
@ll = global i64 10737418300, align 8
@f = global float 1.000000e+00, align 4
@d = global double 1.000000e+00, align 8
@ld = global double 1.000000e+00, align 8
@p = global i8* @c, align 2
```

**Comment by mskvortsov** [comment]
A handcrafted IR without explicit `align` attributes gets translated into an elf with the alignments fully corresponding to the `target datalayout` string and, therefore, EABI:
```
$ cat align.ll 
target datalayout = "e-m:e-p:16:16-i32:16-i64:16-f32:16-f64:16-a:8-n8:16-S16"
target triple = "msp430-elf"
@space0 = global i8 255
@x = global i64 12297829382473034410 ; aaaaaaaaaaaaaaaa
@space1 = global i8 255
@space2 = global i8 255
@space3 = global i8 255
@y = global i32 3149642683           ; bbbbbbbb
@space4 = global i8 255
@space5 = global i8 255
@space6 = global i8 255
@z = global i16 52428                ; cccc
@space7 = global i8 255
@w = global i8  221                  ; dd
$ ~/llvm-build/bin/llc -filetype=obj align.ll 
$ ~/llvm-build/bin/llvm-objdump -s -j .data align.o

align.o:	file format ELF32-msp430

Contents of section .data:
 0000 ff00aaaa aaaaaaaa aaaaffff ff00bbbb  ................
 0010 bbbbffff ff00cccc ffdd               ..........
```
So, the problem is on the frontend side.

**Comment by asl** [comment]
Yep. Just change the alignments of variables in the frontend. Double check with data layout string (in backend and frontend).

**Comment by chbessonova** [comment]
Yes, fixed by #26
